src/BodyFunctionality/BodyPartGenerator.java

```diff
@@ -0,0 +1,382 @@
+        eye.addFeature( "Sclera: WHITE" );
+        eye.addFeature( "Pupil: BLACK" );
+
+        return eye;
+    }
+
+    static BodyPart createEar(String side, AnimalType animalType, Color color)
+    {
+        BodyPart ear = new BodyPart();
+        ear.setName( side + " Ear" );
+        ear.setType( BodyPartType.EAR );
+        ear.setColor( color );
+        ear.setTexture(getAnimalTexture(animalType));
+        ear.setAnimalType( animalType );
+        ear.addFeature( "Shape: Round" );
+
+        return ear;
+    }
+
+    static BodyPart createMuzzle(AnimalType animalType, Color color)
+    {
+        BodyPart muzzle = new BodyPart(  );
+        muzzle.setName( "Muzzle" );
+        muzzle.setType( BodyPartType.MUZZLE );
+        muzzle.setColor( color );
+        muzzle.setTexture(getAnimalTexture(animalType));
+        muzzle.setAnimalType( animalType );
+
+        muzzle.attachedBodyParts.add( createNose(animalType, color));
+        muzzle.attachedBodyParts.add( createMouth(animalType));
+
+        attachBodyParts(muzzle.attachedBodyParts, muzzle);
+
+        return muzzle;
+    }
+
+    static BodyPart createNose(AnimalType animalType, Color color)
+    {
+        BodyPart nose = new BodyPart(  );
+        nose.setName( "Nose" );
+        nose.setType( BodyPartType.NOSE );
+        nose.setColor( color );
+        nose.setTexture(getAnimalTexture(animalType));
+        nose.setAnimalType( animalType );
+        nose.addFeature( "Shape: Round" );
+
+        return nose;
+    }
+
+    static BodyPart createMouth(AnimalType animalType)
+    {
+        BodyPart mouth = new BodyPart(  );
+        mouth.setName( "Mouth" );
+        mouth.setType( BodyPartType.MOUTH );
+        mouth.setColor( Color.RED );
+        mouth.setTexture(getAnimalTexture(animalType));
+        mouth.setAnimalType( animalType );
+
+        mouth.attachedBodyParts.add( createTongue(animalType));
+
+        attachBodyParts(mouth.attachedBodyParts, mouth);
+
+        return mouth;
+    }
+
+    static BodyPart createTongue(AnimalType animalType)
+    {
+        BodyPart tongue = new BodyPart(  );
+        tongue.setName( "Tongue" );
+        tongue.setType( BodyPartType.TONGUE );
+        tongue.setColor( Color.RED );
+        tongue.setTexture(getAnimalTexture(animalType));
+        tongue.setAnimalType( animalType );
+
+        return tongue;
+    }
+
+
+    static void attachBodyParts(ArrayList<BodyPart> parts, BodyPart above)
+    {
+        for (int i=0; i<parts.size(); i++)
+        {
+            parts.get( i ).setAboveBodyPart( above );
+        }
+    }
+
+
+
+    /**
+     * Method that returns the general type of texture for a creature
+     * todo When a new creature is added, list it under the appropriate switch statement
+     *
+     * @param animalType AnimalType: The general type of creature
+     * @return Texture: The texture that that general type of creature has
+     */
+    static Texture getAnimalTexture(AnimalType animalType)
```

Comment:
I don't think we are going to need texture mapping like this - the client will probably end up mapping the textures at runtime. We want the server to keep track of as little data as possible, and send the client as little data as possible, so if it can be generated, it'll be generated then.